addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -739,13 +761,67 @@ QUnit.test('use a command', async function (assert) {
     );
 });
 
+QUnit.test("commands are not displayed when channel types don't match", async function (assert) {
+    assert.expect(3);
+
+    this.data['mail.channel'].records.push({ channel_type: 'chat', id: 20 });
+    this.data['mail.channel_command'].records.push(
+        {
+            channel_types: ['livechat'],
+            help: "See 15 last visited pages",
+            name: "history",
+        },
+    );
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel',
+    });
+    await this.createComposerComponent(thread.composer);
+    assert.containsNone(
+        document.body,
+        '.o_ComposerSuggestionList_list',
+        "command suggestions list should not be present"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_ComposerTextInput_textarea').value,
+        "",
+        "text content of composer should be empty initially"
+    );
```

Comment:
Remove these assertions. We motivated in the past to put premise assertions to workaround the lack of dependencies among QUnit tests, but now I think this is a mistake, and it's best to have very small test even with this lacking feature of QUnit.

Reply:
This is fine, but just be careful to not go the other way too much.
Assertion of initial state can be important to ensure the test actually did something. Makes debugging a lot easier to know the initial state was as expected too.

Reply:
Yes, assert before when a transition is relevant in the test. Just avoid assertion premises that act like _"please crash if test dependencies actually not satisfied"_.

Reply:
_(just to clarify, I agree it sucks having tests that pass when they shouldn't due to dependencies not being satisfied. Just that the workaround of putting more assertions was/is a bad idea, and for the time being it's best keeping test short and straight to the point)_

Reply:
So ?

Reply:
IMO just remove them. Considering name of test, I expect this test having this shape:
```js
// setup
// user attempts making command on channel that hasn't this command
// assert the command is not suggested
```

I wouldn't care having assertions before user interactions. These are noise IMO. Also the resulting assertion should specifically tell this command is not suggested. It should not eliminate possibility that some other commands may be suggested.

Reply:
Ok, I'll wait for @seb-odoo to confirm that decision to do it (in order to avoid make things twice)

Reply:
I'm ok if we remove this one.